ratis-grpc/src/main/java/org/apache/ratis/grpc/server/GrpcLogAppender.java

```diff
@@ -390,6 +390,9 @@ public void onNext(AppendEntriesReplyProto reply) {
       AppendEntriesRequest request = pendingRequests.remove(reply);
       if (request != null) {
         request.stopRequestTimer(); // Update completion time
+        getFollower().updateLastAppendEntriesResponseTime(request.getSendTime()); // Update the last rpc time
+      } else {
+        getFollower().updateLastRpcResponseTime();
       }
```

Comment:
How about moving the `updateLastRpcResponseTime()` out?
```java
        getFollower().updateLastRespondedAppendEntriesSendTime(request.getSendTime());
      }
      getFollower().updateLastRpcResponseTime();
```

Reply:
done.